### PR TITLE
feat: adding management command to remove expired pending group memberships

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
https://2u-internal.atlassian.net/browse/ENT-8524

In this PR:
- Updated EnterpriseCustomerUser readonly serializer to pull group UUIDs instead of membership records
- Updated EnterpriseCustomerUser readonly serializer to account for groups that apply to all contexts
- Updated EnterpriseGroup `/learners/` endpoint to account for groups that apply to all contexts
- New ``applies_to_all_contexts`` flag on EnterpriseGroup records
- New management command ``remove_expired_pending_group_memberships`` that will remove pending membership records if they are older than 90 days
- Updated factories
- Associated unit tests
- Fix to the enterprise UserFactory class to avoid collisions in test data

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/openedx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/openedx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/openedx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/management/commands/remove_expired_pending_group_memberships.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        $ ./manage.py remove_expired_pending_group_memberships
+    """
+    help = 'Removes pending group memberships if they are older than 90 days.'
+
+    def add_arguments(self, parser):
+        parser.add_argument("-e", "--enterprise_custom")
+
+    def handle(self, *args, **options):
+        queryset = EnterpriseGroupMembership.objects.all()
+        if enterprise_arg := options.get("enterprise_customer"):
```
Comment (marlonkeating) [inline enterprise/management/commands/remove_expired_pending_group_memberships.py:32]: Are we documenting the optional `enterprise_customer` argument somewhere?

File: enterprise/management/commands/remove_expired_pending_group_memberships.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    Management command for ensuring any pending group membership, ie memberships associated with a pending enterprise
+    user, are removed after 90 days.
+
+    Example usage:
+        $ ./manage.py remove_expired_pending_group_memberships
+    """
+    help = 'Removes pending group memberships if they are older than 90 days.'
+
+    def add_arguments(self, parser):
+        parser.add_argument("-e", "--enterprise_custom")
```
Comment (marlonkeating) [inline enterprise/management/commands/remove_expired_pending_group_memberships.py:28]: typo?
```suggestion
        parser.add_argument("-e", "--enterprise_customer")
```

Comment (alex-sheehan-edx) [reply enterprise/management/commands/remove_expired_pending_group_memberships.py:28]: 😂 YIKES good catch haha

File: enterprise/management/commands/remove_expired_pending_group_memberships.py (hunk 1 above)
Comment (alex-sheehan-edx) [reply enterprise/management/commands/remove_expired_pending_group_memberships.py:32]: uhhh Let me add it to the doc string, it's not necessary for the overnight job that's to run however I liked the added flexibility for other users down the line

Comment (alex-sheehan-edx) [reply enterprise/management/commands/remove_expired_pending_group_memberships.py:33]: updated the doc string- I'll write a quick unit test for it just to be certain

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**